### ISSUE TITLE
Screens display in reverse order when passcode is needed

### ISSUE THREAD

**Issue by clarmso** [mozilla-mobile/firefox-ios#15642]
## Steps to reproduce
1. Restart the iOS simulator.
2. Open Firefox iOS.
3. Tap hamburger menu. Select "Passwords" from the menu.

### Expected behavior
The "Your passwords are now protected by Face ID" screen should appear before the "Enter iPhone passcode" page.

### Actual behavior
We need to enter the passcode first before seeing the "Your passwords are now protected by Face ID" screen.

### Device & build information
* Device: iPhone 14 simulator
* Build version: 117.0 (1) July 18 2023
* First seen version: Last 2-3 weeks?
### Notes
Attachments: 

https://github.com/mozilla-mobile/firefox-ios/assets/1740517/42f66757-15d6-49bf-82f9-940a00ee2fde

Note that the XCUITests indicate that the "Your passwords are now protected by Face ID" screen should appear *before* the passcode is entered.

**Comment by clarmso** [comment]
Oops. Duplicate bug. I missed the discussion. https://github.com/mozilla-mobile/firefox-ios/issues/15361

**Comment by clarmso** [comment]
Hmm...In #15361, I said that I could not get out of the password page. I can now get out of the passwords page, but the the passcode page should appear *after* "Your passwords are now protected by Face ID" screen.

I'm not 100% sure if this comment says that the new behaviour should be accepted.
https://github.com/mozilla-mobile/firefox-ios/issues/15361#issuecomment-1638287209

**Comment by isabelrios** [comment]
I think this is the issue I was commenting in #15509. Although I think there are actually two issues when accessing Passwords menu for **the first time**:

1. The one dectected by the automated test `testCreateLoginManually`:
- Go to Settings -> Passwords 
_Actual_: the locked screen is shown asking for your code for a few seconds. Then, the biometic informative screen is shown, once you click on Continue, the locked screen is shown again.
_Expected_: the biometric screen is shown and then the locked screen.

2. The one dectected by the l10n automated test `testLoginDetails`:
- Go to Browser Tab Menu -> Passwords
_Actual_: The biometric screen is shown
_Expected_: The locked screen is shown and logins can be accessed without having shown the biometric informative screen

@lmarceau @dnarcese Both tests expect the biometric informative screen to be shown the first time user enters the Passwords menu. 
If the order of the screens there have changed, or there is a new expected behaviour with those screens, please let us know so that we can update the tests accordingly. Thanks!

**Comment by clarmso** [comment]
@isabelrios Please feel free to close this issue if it's a duplicate of https://github.com/mozilla-mobile/firefox-ios/issues/15509. 🙂 Yes, the XCUITest confirmed the existing order of those screen.

**Comment by isabelrios** [comment]
I do not think this is a duplicate, the cause could be related, but this is a different issue.

**Comment by isabelrios** [comment]
@dnarcese We would need to know the expected behaviour here and so apply it to the tests or make temporary changes until we have a fix.

One of the tests affected runs in the Smoketest and even though it is not blocking because it passes in a re-try, it causes a bit longer build and all the red check marks on github :( 

Could you please let us know if it is a bug or a new behaviour at least to know how to proceed for now?

**Comment by dnarcese** [comment]
I think that the info screen should show first every time until the user clicks continue.

**Comment by lmarceau** [comment]
@dnarcese looking at the issue, I see why now why it was done that way. We use the info screen for credit card code as well, and we show the locked screen first. So I assume we want the same behavior for credit card, where we would show the info screen before showing the locked screen (as we'll fix this for passwords).

**Comment by lmarceau** [comment]
Mh, I confused the `PasswordManagerOnboardingViewController` page with the `DevicePasscodeRequiredViewController` view. They look the same, but are of course different. My previous comment related to `DevicePasscodeRequiredViewController` which we had some conversation over in https://github.com/mozilla-mobile/firefox-ios/issues/15782 where they should align for CC and passwords. 

That being said, I'll look into fixing the current bug 😄

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Verified as fixed on v117 (33597) with iPhone 13 Pro (15.7.1).

Laurie Marceauas I didn`t see a note for QA, I think for the first time when we visit the section we should have "Your passwords are now protected by Face ID" screen should appear, and then on the next visit of the section, we should always have the "Enter iPhone passcode" page.

Can you confirm the one from above?
